docs/definition.rst

```diff
@@ -4,210 +4,224 @@ Execution Environment Definition
+      Commands to insert before building of the final image.
 
-This section is a dictionary value that is used to define the Ansible Galaxy,
-Python, and system dependencies that must be installed into the final container.
-Valid keys for this section are:
+    ``append_final``
+      Commands to insert after building of the final image.
 
-``galaxy``
-  This string value is the path to a file containing the Ansible Galaxy
-  dependencies to be installed with the ``ansible-galaxy collection install -r ...``
-  command.
+build_arg_defaults
+******************
 
-  The supplied value may be a relative path from the directory of the execution
-  environment definition's folder, or an absolute path.
+Default values for build args can be specified in the definition file in
+the ``build_arg_defaults`` section as a dictionary. This is an alternative
+to using the :ref:`build-arg` CLI flag.
 
-``python``
-  This string value is the path to a file containing the Python dependencies
-  to be installed with the ``pip install -r ...`` command.
+Build args used by ``ansible-builder`` are the following:
 
-  The supplied value may be a relative path from the directory of the execution
-  environment definition's folder, or an absolute path.
+    ``ANSIBLE_GALAXY_CLI_COLLECTION_OPTS``
+      This allows the user to pass the '--pre' flag (or others) to enable the installation of pre-releases collections.
 
-``system``
-  This string value is points to a
-  `bindep <https://docs.openstack.org/infra/bindep/readme.html>`__
-  requirements file. This will be processed by ``bindep`` and then passed
-  to ``dnf``, other platforms are not yet supported.
+    ``ANSIBLE_GALAXY_CLI_ROLE_OPTS``
+      This allows the user to pass any flags to the role installation.
 
-additional_build_steps
-^^^^^^^^^^^^^^^^^^^^^^
+Values given inside of ``build_arg_defaults`` will be hard-coded into the
+Containerfile, so they will persist if ``podman build`` is called manually.
 
-Additional commands may be specified in the ``additional_build_steps``
-section, either for before the main build steps (``prepend``) or after
-(``append``). The syntax needs to be one of the following:
+If the same variable is specified in the CLI :ref:`build-arg` flag,
+the CLI value will take higher precedence.
 
-- a multi-line string (example shown in the ``prepend`` section above)
-- a list (as shown via ``append``)
+dependencies
+************
 
-.. _version-2:
+This section allows you to describe any dependencies that will need to be
+installed into the final image.
 
-Version 2 Format
-----------------
+The following keys are valid for this section:
 
-With the version 2 format, an execution environment definition may specify
-a base and builder container image whose signature must be validated before
-builder will build the resulting image, based on the value of the
-:ref:`container-policy` CLI option.
+    ``ansible_core``
+      The version of the Ansible python package to be installed by pip.
```

Comment:
Should probably mention that whatever's here for both `ansible_core` will get passed directly to `pip install`, so it can be a local/http/git path, an arbitrary package name with or without a version, etc, and maybe some examples, eg:
`ansible_core: ansible-core` <-- latest from PyPI (or whatever the configured pip index is on the builder host)
`ansible_core: ansible-core==2.14.2` <-- exact ansible_core version from PyPI/configured index
`ansible_core: https://github.com/nitzmahone/ansible/archive/refs/heads/tempfile_secontext.tar.gz` <-- install from a GitHub PR tarball